BDArmory/Weapons/Missiles/BDModularGuidance.cs

```diff
@@ -778,10 +820,192 @@ private Vector3 CruiseGuidance()
+                commander.controlPoints.Add("dynamic", dynamic);
+            }
+
+            commander.SetControlPoint("dynamic");
+
+            Vector3 referenceRoll = commander.part.GetReferenceTransform().forward;
+            Vector3 roll = referenceRoll != direction.normalized ? referenceRoll : commander.transform.forward;
+
+            // Orient the control point towards direction (finger) with perpendicular as the up vector (thumb).
+            Vector3 perpendicular = Vector3.ProjectOnPlane(roll, direction.normalized);
+            dynamic.transform.rotation = Quaternion.LookRotation(perpendicular, direction.normalized); // VAB orientation.
+        }
+
+        private static Vector3 GetFireVector(List<ModuleEngines> engines, List<ModuleRCSFX> RCS = null, Vector3 thrustVector = default(Vector3))
+        {
+            // Place linears first to establish a direction, not currently needed
+            //RCS.Sort((a, b) => a.thrusterTransforms.Count().CompareTo(b.thrusterTransforms.Count()));
+
+            if (engines?.Any() == true)
+            {
+                // If there are engines we can override any potential provided vector
+                thrustVector = GetMeanVector(engines.First());
+                foreach (ModuleEngines engine in engines.Skip(1))
+                {
+                    thrustVector += GetMeanVector(engine);
+                }
+                if (RCS?.Any() == true)
+                {
+                    // If there are engines we can add RCS on top
+                    foreach (ModuleRCSFX thruster in RCS)
+                    {
+                        thrustVector += GetRCSVector(thruster, thrustVector);
+                    }
+                }
+            }
+            else if (RCS?.Any() == true)
+            {
+                // If there are no engines we have to plot the RCS along the provided vector
+                Vector3 rcsVector = GetRCSVector(RCS.First(), thrustVector);
+                foreach (ModuleRCSFX thruster in RCS.Skip(1))
+                {
+                    rcsVector += GetRCSVector(thruster, thrustVector);
+                }
+                //replace thrustVector with RCS Vector
+                thrustVector = rcsVector;
+            }
+
+            return thrustVector;
+        }
+
+        private static Vector3 GetRCSVector(ModuleRCSFX thruster, Vector3 thrustVector)
+        {
+            //method to get the thrust vector of a specified rcs thruster
+            List<Transform> positions = thruster.thrusterTransforms;
+            Vector3 meanVector = Vector3.zero;
+
+            foreach (Transform thrusterTransform in positions)
+            {
+                Vector3 pos = thrusterTransform.up * thruster.thrusterPower;
+                // rcs will fire if thrust goes in the forward direction by any degree, this is reduced with angle offset
+                if (Vector3.Dot(thrustVector.normalized, pos.normalized) > 0)
```

Comment:
Since the dot product is also used below, it'd be better to compute it once to avoid two extra normalizations.